contrib/utilities/detect_include_cycles.py

```diff
@@ -47,11 +47,13 @@ def add_includes_for_file(header_file_name, G) :
 
 
 
-# Take the list of call arguments, excluding the first that contains
-# the name of the executable (i.e., this program). For each, add the
-# includes as the edges of a directed graph.
+# Create a list of all header files in the include folder.
+filelist = glob("include/deal.II/*/*.h")
```

Comment:
Actually, I later realized that there are also header files in further *sub*directories, e.g., `include/deal.II/physics/elasticity/standard_tensors.h`. Can you search recursively for all `*.h` files in `include/`?

Reply:
I have added the recursion feature to `glob`. No more cycles were detected.